src/puppetlabs/puppetdb/command.clj

```diff
@@ -280,29 +233,33 @@
                id (command-names :store-report)
                puppet_version certname)))
 
-(defmethod process-command! [(command-names :store-report) 3]
-  [command db]
-  (store-report* 6 db
-                 (let [received-time (get-in command [:annotations :received])]
-                   (update command :payload report/wire-v3->wire-v6 received-time))))
+;; ## Command processors
 
-(defmethod process-command! [(command-names :store-report) 4]
-  [command db]
-  (store-report* 6 db
-                 (let [received-time (get-in command [:annotations :received])]
-                   (update command :payload report/wire-v4->wire-v6 received-time))))
+(def supported-command?
+  (comp (kitchensink/valset command-names) :command))
 
-(defmethod process-command! [(command-names :store-report) 5]
-  [command db]
-  (store-report* 6 db
-                 (update command :payload report/wire-v5->wire-v6)))
+(defn supported-version? [command version]
+  (let [[min-version max-version] (get supported-command-versions command)]
+    (<= min-version version max-version)))
 
-(defmethod process-command! [(command-names :store-report) 6]
-  [command db]
-  (store-report* 6 db command))
+(defn process-command!
```

Comment:
@senior I was curious what you though about this, I got rid of the defmethod entirely